### ISSUE TITLE
Question about the version of DL3DV

### ISSUE THREAD

**Issue by HLinChen** [CUT3R/CUT3R#23]
Hi! Thanks for your great work.

When I want to download and process the DL3DV dataset, I find there are several versions of the dataset, like 480P, 960P, 2K, and so on. Could you tell me which version I should download? Thank you for your help!

Looking forward to your reply.

**Comment by qianqianwang68** [comment]
Hi @HLinChen,

We used the 960P resolution for DL3DV. However, we just realized that due to the undistortion process in COLMAP, the depth maps and masks we computed correspond to the undistorted images, which may have slight differences in resolution compared to DL3DV’s original images. A simple but suboptimal solution is to resize our processed depth maps and masks to match DL3DV’s original images. We are also working on releasing our undistorted images, but we need to coordinate with DL3DV’s team to ensure compliance with their terms of use. Please stay tuned.

**Comment by HLinChen** [comment]
Thank you for your reply! Looking forward to your release!

**Comment by NIRVANALAN** [comment]
Hi, any updates on this thread? Thanks in advance!

**Comment by Forrest-110** [comment]
Hi @NIRVANALAN ! We have reached an agreement on the release plan. Please stay tuned!

**Comment by NIRVANALAN** [comment]
Great, looking forward to the release.

**Comment by RahulSajnani** [comment]
Hi @Forrest-110 ! Thank you for your great work. I have some concerns about the depth map extraction from DL3DV. I tried to obtain depth maps for DL3DV but the colmap produced geometric depth maps have too many outliers. Can you please detail the steps that you take to get clean DL3DV depth maps or outlier masks?

**Comment by RahulSajnani** [comment]
@Forrest-110 Thank you so much! I will get back to you if I have any other concerns

**Comment by RahulSajnani** [comment]
Hi @Forrest-110 so it seems like with this preprocessing of depth maps based on large gradients still have a lot of smearing artefacts as well as removes some important objects within the scene. 
I am wondering if you faced the same issue. If so, how does CUT3R produce such good results when the training data has so many depth artefacts? Is the confidence value helping in ignoring bad depths automatically?

**Comment by Forrest-110** [comment]
Hi @RahulSajnani! Are you using the depth maps we released for DL3DV? They should be clean enough when filtered using the script above. And, yes—the confidence values can automatically handle noisy points.

**Comment by RahulSajnani** [comment]
Hi @Forrest-110, I have tried to download the depth maps provided by you a lot of times both using gdown as well as web and it always fails because of some google quota restrictions. Hence, I ran colmap on 1K split of DL3DV. The script you provided does filter out points, but they also filter out some important objects in the scene for example bushes infront of camera.

**Comment by Forrest-110** [comment]
Hi @RahulSajnani! We use a conservative filtering strategy, since retaining noisy points is more harmful than discarding potentially meaningful ones.

**Comment by NIRVANALAN** [comment]
Hi, so you use a filtered version of DL3DV data right, may I ask how much did you drop statistically? By the way, any ETA of the rgb image release ;)

**Comment by Forrest-110** [comment]
Hi @NIRVANALAN! We mask out 20 to 50 % of points per depthmap. This is only a rough estimate.

As for the ETA of the DL3DV images, we're still working on it, but we expect to release them before May. Please stay tuned!

**Comment by NIRVANALAN** [comment]
Thanks for your prompt reply. Thanks and looking forward to the data release soon!

**Comment by jedyang97** [comment]
> Hi [@HLinChen](https://github.com/HLinChen),
> 
> We used the 960P resolution for DL3DV. However, we just realized that due to the undistortion process in COLMAP, the depth maps and masks we computed correspond to the undistorted images, which may have slight differences in resolution compared to DL3DV’s original images. A simple but suboptimal solution is to resize our processed depth maps and masks to match DL3DV’s original images. We are also working on releasing our undistorted images, but we need to coordinate with DL3DV’s team to ensure compliance with their terms of use. Please stay tuned.

Thank you for preparing the data for the community. This is a very helpful and generous thing for you to do! I was wondering, while we wait for the RGB image release, would it be possible if you release a simple script that you use to undistort the 960P image from DL3DV official download? With this script, we could also try to produce RGB images that matches the depth/outlier_mask ourselves. Thank you!

**Comment by jedyang97** [comment]
OK that would be great! Looking forward to it!

**Comment by jedyang97** [comment]
To facilitate your release, here is a script I typically use to split and compress large data folders into individually decompressable.zips in a multi-process parallel manner: https://gist.github.com/jedyang97/e6bf805213898aed9f81948807e48b8e This will make it easier to upload large data to HuggingFace datasets.

I hope this can help your release of DL3DV data!

**Comment by Forrest-110** [comment]
We have released the full version of our processed DL3DV. Please refer to [here](https://github.com/CUT3R/CUT3R/blob/main/docs/preprocess.md#dl3dv) for more details.

**Comment by Gynjn** [comment]
@Forrest-110 
Thanks for sharing great datasets.
BTW, the camera convention is blender or opencv?

**Comment by Forrest-110** [comment]
Hi @Gynjn ! The camera convention is opencv, and pose is c2w (the relative pose from camera coordinate to world coordinate).

**Comment by NIRVANALAN** [comment]
Hi, thanks for releasing the dl3dv data.

When I am checking the data, I find like folders like ```7K/9b75f3bcdfb2fa943d151e8db52aa597b2ebb47caa80c056c268a282c7bc4d94/dense/depth``` have another "depth" folder ```7K/9b75f3bcdfb2fa943d151e8db52aa597b2ebb47caa80c056c268a282c7bc4d94/dense/depth/depth/ ```, which should be used instead (has the same amount of data as the outlier_mask, sky_mask folder). Any ideas?

**Comment by Forrest-110** [comment]
Hi @NIRVANALAN ! Thank you for pointing that out! Upon initial review, these two folders appear to contain the same content, but we'll verify this thoroughly.

In the meantime, we recommend using the content from:
`7K/9b75f3bcdfb2fa943d151e8db52aa597b2ebb47caa80c056c268a282c7bc4d94/dense/depth`

We’ll confirm and clean up the data.

**Comment by NIRVANALAN** [comment]
Thanks for your reply. I noticed the ```7K/9b75f3bcdfb2fa943d151e8db52aa597b2ebb47caa80c056c268a282c7bc4d94/dense/depth``` contains less images than ```7K/9b75f3bcdfb2fa943d151e8db52aa597b2ebb47caa80c056c268a282c7bc4d94/dense/valid_mask```, while the ```7K/9b75f3bcdfb2fa943d151e8db52aa597b2ebb47caa80c056c268a282c7bc4d94/dense/depth/depth``` has the same number of images (305). Please let me know if you have any updates.

**Comment by Forrest-110** [comment]
Hi @NIRVANALAN ! We check the data using the following script, 
```
import os
import numpy as np

def compare_content(dir1, dir2):
    # Get directory contents, excluding 'depth' subdirectory in dir1
    content1 = set(os.listdir(dir1)) - {'depth'}
    content2 = set(os.listdir(dir2))
    
    # Check subset relationship
    is_dir1_subset = content1.issubset(content2)
    is_dir2_subset = content2.issubset(content1)
    
    if not (is_dir1_subset or is_dir2_subset):
        # Identify unique files in both directories
        unique_to_dir1 = content1 - content2
        unique_to_dir2 = content2 - content1
        error_msg = (f"Directory contents are not subsets:\n"
                     f"Unique to {dir1}: {unique_to_dir1 or '∅'}\n"
                     f"Unique to {dir2}: {unique_to_dir2 or '∅'}")
        raise AssertionError(error_msg)
    
    # Determine which files to compare (the smaller set)
    compare_files = content1 if is_dir1_subset else content2
    
    # Compare each common file
    for filename in compare_files:
        file1 = os.path.join(dir1, filename)
        file2 = os.path.join(dir2, filename)
        
        try:
            arr1 = np.load(file1)
            arr2 = np.load(file2)
        except Exception as e:
            print(f"Failed to load {filename}: {e}")
            
        if not np.allclose(arr1, arr2):
            print(f"Data mismatch in {filename}")


root = 'custom_data/processed_dl3dv'
seqs = os.listdir(root)
subseqs = []

# Gather valid dense subdirectories
for seq in seqs:
    seq_path = os.path.join(root, seq)
    if not os.path.isdir(seq_path):
        continue
    for entry in os.listdir(seq_path):
        entry_path = os.path.join(seq_path, entry)
        if os.path.isdir(entry_path):
            dense_path = os.path.join(entry_path, 'dense')
            if os.path.isdir(dense_path):
                subseqs.append(dense_path)

subseqs.sort()

# Validate depth directories
for subseq in subseqs:
    depth_dir = os.path.join(subseq, 'depth')
    depth_subdir = os.path.join(depth_dir, 'depth')
    
    if os.path.isdir(depth_subdir):
        print(f"Comparing {depth_dir} and {depth_subdir}")
        try:
            compare_content(depth_dir, depth_subdir)
            print("All files match!\n")
        except Exception as e:
            print(f"Validation failed: {e}\n")
```

And it looks like only `custom_data/processed_dl3dv/7K/9b75f3bcdfb2fa943d151e8db52aa597b2ebb47caa80c056c268a282c7bc4d94/dense/depth` has such issue:
```
Comparing custom_data/processed_dl3dv/7K/9b75f3bcdfb2fa943d151e8db52aa597b2ebb47caa80c056c268a282c7bc4d94/dense/depth and custom_data/processed_dl3dv/7K/9b75f3bcdfb2fa943d151e8db52aa597b2ebb47caa80c056c268a282c7bc4d94/dense/depth/depth
Failed to load frame_00219.npy: cannot reshape array of size 475104 into shape (535,953)
```
Please use `7K/9b75f3bcdfb2fa943d151e8db52aa597b2ebb47caa80c056c268a282c7bc4d94/dense/depth/depth`, or simply skip this sequence.

**Comment by NIRVANALAN** [comment]
Thanks for your prompt reply, and I will follow your suggestions ;)

**Comment by jedyang97** [comment]
Hi @Forrest-110, thank you so much for releasing the RGB data! May I know what data is contained in [CUT3R_release/processed_dl3dv_ours_parts](https://huggingface.co/datasets/zhangify/CUT3R_release/tree/main), and what is the format / how to decompress those files? I was hoping this is replacement of the 3TB Google Drive file, which is a bit hard to download. Thanks!

**Comment by Forrest-110** [comment]
Hi @jedyang97 ! Sorry for the late reply. The data includes everything from the processed DL3DV data except for the images and camera info. To extract these files, please use the command: `cat processed_filename_part_* | tar -xzvf -` and replace the filename.

**Comment by Forrest-110** [comment]
Hi @smileyenot983 ! I don't quite understand your question. Could you provide more details about it? like what does '3K' mean and what is the merge script?

**Comment by smileyenot983** [comment]
> Hi [@smileyenot983](https://github.com/smileyenot983) ! I don't quite understand your question. Could you provide more details about it? like what does '3K' mean and what is the merge script?

Hello! I have confused a bit merge script and preprocess script, i meant preprocess in previous comment

In dl3dv dataset which was released on hugging face there are multiple folders, like 3K,5K etc. In preprocess script(https://github.com/CUT3R/CUT3R/blob/main/datasets_preprocess/preprocess_dl3dv.py#L29) 3K folder is chosen

**Comment by Forrest-110** [comment]
Hi @smileyenot983 ! This preprocessing script should be run separately for each `xK` folder. For example, if you downloaded `5K` and `6K`, modify `3K` in the script to `5K` and run it, then change it to `6K` and run it again.
